test/e2e/prometheus_test.go

```diff
@@ -92,3 +97,180 @@ func TestPrometheusAlertmanagerAntiAffinity(t *testing.T) {
 		t.Fatal("Can not find pods: prometheus-k8s or alertmanager-main")
 	}
 }
+
+func TestPrometheusRemoteWrite(t *testing.T) {
+	ctx := context.Background()
+
+	targetDeployment := `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: instrumented-sample-app
+  namespace: openshift-monitoring
+  labels:
+    group: test
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      group: test
+  template:
+    metadata:
+      labels:
+        group: test
+    spec:
+      containers:
+      - name: example-app
+        args:
+        - --cert-path=/etc/certs
+        image: quay.io/coreos/instrumented-sample-app:0.2.0-bearer-mtls-1
+        imagePullPolicy: IfNotPresent
+        ports:
+        - name: web
+          containerPort: 8080
+        - name: mtls
+          containerPort: 8081
+        volumeMounts:
+	- mountPath: /etc/certs
+	  name: certs
+        - mountPath: /etc/certs/tls
+          name: rw-target-tls
+        - mountPath: /etc/certs/ca
+          name: serving-certs-ca-bundle
+      volumes:
+      - name: rw-target-tls
+        secret:
+          secretName: rw-target-tls
+      - name: serving-certs-ca-bundle
+        configMap:
+	  name: serving-certs-ca-bundle
+      - name: certs
+        emptyDir: {}
+      initContainers:
+      - command:
+        - /bin/sh
+        - -c
+        - 'cp /etc/certs/tls/tls.key /etc/certs/key.pem; cat /etc/certs/tls/tls.crt > /etc/certs/cert.pem; cat /etc/certs/ca/service-ca.crt >> /etc/certs/cert.pem'
+	name: crypt-init
+	image: busybox
+	volumeMounts:
+	- mountPath: /etc/certs
+	  name: certs
+        - mountPath: /etc/certs/tls
+          name: rw-target-tls
+        - mountPath: /etc/certs/ca
+          name: serving-certs-ca-bundle
+`
+	deployment := &appsv1.Deployment{}
+	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(targetDeployment)), 100).Decode(&deployment); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
nit: we could reuse https://github.com/openshift/cluster-monitoring-operator/blob/master/pkg/manifests/manifests.go#L3680-L3688